### PR TITLE
Fixes trauma healing nanites.

### PR DESCRIPTION
**Pull request by Arturlang**
Forgot to fix them in my last nanite fix PR, is ported from TG. Has possible balance implications since it makes brain/trauma healing nanites way better.
:cl:
fix: Fixes brain damage/trauma healing nanites so they actually work while there are only traumas.
/:cl:

### CODE REVIEW

File: code/modules/research/nanites/nanite_programs/healing.dm [hunk 1]
```diff
@@ -70,9 +70,14 @@
 	rogue_types = list(/datum/nanite_program/brain_decay)
 
 /datum/nanite_program/brain_heal/check_conditions()
-	if(!host_mob.getBrainLoss())
-		return FALSE
-	return ..()
+	var/problems = FALSE
```
Comment (deathride58) [inline code/modules/research/nanites/nanite_programs/healing.dm]: Eugh. What happened to TG's coding standards?
You can axe this var declaration then replace the `problems = TRUE` lines with `return ..()`, and replace the final line in the proc with a simple `return FALSE`
There's no such thing as enough micro-optimizations, especially for things like nanites that get their procs called repeatedly en-masse. Var declarations within procs in byond have a notable performance impact, so you typically want to keep var declarations to a minimum whenever possible inside procs that get called frequently.

**Review by deathride58 — CHANGES_REQUESTED** [review]

File: code/modules/research/nanites/nanite_programs/healing.dm (hunk 1 above)
Comment (Arturlang) [reply code/modules/research/nanites/nanite_programs/healing.dm]: I guess oranges truly ruined em good.

Comment (Arturlang) [reply code/modules/research/nanites/nanite_programs/healing.dm]: I wish i knew how to these optimizations myself honestly

**Comment by Arturlang** [comment]
Wierd how it still says changes needed when i did those changes and the comment is resolved

**Review by deathride58 — APPROVED** [review]

**Pull request merged**